### PR TITLE
findutils: Ensure __nonnull is defined

### PR DESCRIPTION
**Pull request by eschnett**
This cures a build error with gcc on macOS.

### CODE REVIEW

**Comment by haampie** [comment]
@spackbot run pipeline

**Comment by alalazo** [comment]
Can you point to some bug report or description of the underlying issue? Technically iirc defining `__` macro triggers UB, and what I wonder is what `__nonnull` is expected to return / be expanded to.

**Comment by eschnett** [comment]
I opened a PR at https://github.com/spack/spack/issues/28356.

File: var/spack/repos/builtin/packages/findutils/package.py [hunk 1]
```diff
@@ -51,6 +51,7 @@ def url_for_version(self, version):
     patch('nvhpc.patch', when='@4.6.0 %nvhpc')
     # Workaround bug where __LONG_WIDTH__ is not defined
     patch('nvhpc-long-width.patch', when='@4.8.0:4.8 %nvhpc')
+    patch('nonnull.patch')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/findutils/package.py:58]: Thanks for the explanation, can you add a comment above this patch directive with a reference to https://savannah.gnu.org/bugs/?func=detailitem&item_id=59972 and a very brief description of the issue it solves?

Comment (eschnett) [reply var/spack/repos/builtin/packages/findutils/package.py:58]: Done.

**Comment by alalazo** [comment]
Can you rebase to run Gitlab pipelines on a new merge commit?

**Comment by eschnett** [comment]
@spackbot run pipeline

**Review by alalazo — APPROVED** [review]

**Comment by climbfuji** [comment]
@eschnett This patch seems to break the findutils build on my macOS. I am using spack commit c791ddc742474226bed94fcfadd5965148957918 from Feb 7. My setup is macOS Monterey 12.1, running in Rosetta 2 x86_64 emulator mode, using llvm-clang@13.0.0 installed via homebrew. This is the error I got:
```
==> Installing findutils-4.7.0-h3fhihbyipvpkpyosc3v3bseblc4ak3s
==> No binary for findutils-4.7.0-h3fhihbyipvpkpyosc3v3bseblc4ak3s found: installing from source
/Users/heinzell/work/jedi-stack/spack-stack-new-joint/spack-stack-develop-20220209/spack/lib/spack/spack/target.py:137: UserWarning: microarchitecture specific optimizations are not supported yet on mixed compiler toolchains [check clang@13.0.0 for further details]
  warnings.warn(msg.format(compiler))
The text leading up to this was:
--------------------------
|--- a/gl/lib/malloc/dynarray-skeleton.c
|+++ b/gl/lib/malloc/dynarray-skeleton.c
--------------------------
File to patch:
Skip this patch? [y]
1 out of 1 hunk ignored
==> Fetching https://mirror.spack.io/_source-cache/archive/c5/c5fefbdf9858f7e4feb86f036e1247a54c79fc2d8e4b7064d5aaa1f47dfa789a.tar.xz
==> Patch /Users/heinzell/work/jedi-stack/spack-stack-new-joint/spack-stack-develop-20220209/spack/var/spack/repos/builtin/packages/findutils/nonnull.patch failed.
==> Error: ProcessError: Command exited with status 1:
    '/usr/bin/patch' '-s' '-p' '1' '-i' '/Users/heinzell/work/jedi-stack/spack-stack-new-joint/spack-stack-develop-20220209/spack/var/spack/repos/builtin/packages/findutils/nonnull.patch' '-d' '.'
```
Am I doing something wrong?

**Comment by kgerheiser** [comment]
`gl/lib/malloc` does not exist in findutils < 4.8.0

**Comment by climbfuji** [comment]
> `gl/lib/malloc` does not exist in findutils < 4.8.0

Thanks, @kgerheiser. I'll submit a bugfix PR later today.

**Comment by climbfuji** [comment]
I created an issue for this, and a PR to fix it: https://github.com/spack/spack/issues/28856 and https://github.com/spack/spack/pull/28857

**Pull request merged**